go/tools/builders/compilepkg.go

```diff
@@ -433,7 +433,8 @@ func compileGo(goenv *env, srcs []string, packagePath, importcfgPath, asmHdrPath
 	args = append(args, "--")
 	args = append(args, srcs...)
 	absArgs(args, []string{"-I", "-o", "-trimpath", "-importcfg"})
-	return goenv.runCommand(args)
+	// "go tool compile" writes syntax errors to stdout. Redirecting them to stderr
```

Comment:
Instead of changing this file, let's change `env.runCommand`.

Add a comment that this mirrors behavior in the `go` command: https://go.googlesource.com/go/+/refs/tags/go1.15.2/src/cmd/go/internal/work/exec.go#1958